tools/integration_tests/emulator_tests/streaming_writes_failure/local_file_failure_test.go

```diff
@@ -0,0 +1,302 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package streaming_writes_failure
+
+import (
+	"context"
+	"log"
+	"os"
+	"testing"
+
+	"cloud.google.com/go/storage"
+	emulator_tests "github.com/googlecloudplatform/gcsfuse/v2/tools/integration_tests/emulator_tests/util"
+	"github.com/googlecloudplatform/gcsfuse/v2/tools/integration_tests/util/client"
+	. "github.com/googlecloudplatform/gcsfuse/v2/tools/integration_tests/util/client"
+	"github.com/googlecloudplatform/gcsfuse/v2/tools/integration_tests/util/operations"
+	"github.com/googlecloudplatform/gcsfuse/v2/tools/integration_tests/util/setup"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+// //////////////////////////////////////////////////////////////////////
+// Boilerplate
+// //////////////////////////////////////////////////////////////////////
+
+type defaultFailureTestSuite struct {
+	suite.Suite
+	configPath         string
+	flags              []string
+	testDirPath        string
+	filePath           string
+	fh1                *os.File
+	storageClient      *storage.Client
+	closeStorageClient func() error
+	ctx                context.Context
+}
+
+// //////////////////////////////////////////////////////////////////////
+// Tests
+// //////////////////////////////////////////////////////////////////////
+func (t *defaultFailureTestSuite) SetupSuite() {
```

Comment:
If fileHandle writes results in an error, any filehandle opened after that tries to write it fails and all fileHandles needs to be closed to recover.

If there is a fileHandle opened in read mode, we need not close that to recover. Please add test case for this in seperate PR

Reply:
thats right. We will support reads on unfinalized objects in ZB, so read fileHandles shouldn't effect bwh from recovering.